core/src/main/java/org/infinispan/remoting/inboundhandler/DefaultTopologyRunnable.java

```diff
@@ -39,19 +38,31 @@ public boolean isReady() {
    }
 
    @Override
-   protected Response beforeInvoke() throws Exception {
+   protected InvocationStatus beforeInvoke() throws Exception {
+      // Read commands are executed directly in OOB thread pool. When the many commands are received prior
+      // to the first topology update (when the node is joining), these can wait for transaction data
+      // (even in non-tx mode) and deplete OOB TP => the node stops being responsive event to ST.
```

Comment:
BTW, with ISPN-5469 I'm trying to remove `DefaultTopologyRunnable` altogether and wait (asynchronously) for the topology in the command itself.